### PR TITLE
Add Focus Management section content

### PR DESCRIPTION
**Pull request by glunn**
Content to address: https://github.com/MelSumner/ember-a11y-for/issues/8

Add a section on focus management. It should include:

- [x]  what the browser naturally gives you
- [x]  browsing mode vs focus mode in screen readers
- [x]  focus should return to the place it came from
- [ ]  how focus works in Ember
- [x]  addons to consider

### CODE REVIEW

File: app/templates/index.hbs [hunk 1]
```diff
@@ -172,13 +172,35 @@ id="start-new-app"
...
 <h2 id="section-focus-management">
   <a href="">&sect;</a> Focus Management
 </h2>
-<p>Coming Soon!</p>
---}}
+<p>Focus determines which part of your application receives events from the keyboard and clipboard at any given time. Good focus management can provide a better experience for all of your users, but is particularly critical for those who may navigate exclusively via their keyboard or other input device, such as a switch device.</p>
+
+<h3>Included in the browser</h3>
+<p>Browsers come with built-in focus behavior based on the type of elements used. Interactive elements like buttons and text fields are implicitly focusable, which means that they are automatically inserted into the tab order of the page and come with built-in event handling.</p>
+<p>Non-interactive elements purposely do not receive focus without developer intervention, because no action from the user is expected.</p>
```
[the remark this answers is not in the document]
Comment (glunn) [reply app/templates/index.hbs]: Oh nice, I hadn't realized that the readme for `ember-template-lint` was so helpful. Thanks, going to start referring to this more myself as well!

File: app/templates/index.hbs [hunk 2]
```diff
@@ -172,13 +172,35 @@ id="start-new-app"
...
+<p>It's not possible, however, to interact with the page while in browse mode. Instead, users can switch to focus mode which will pass keystrokes directly to the browser. Navigation in focus mode behaves like keyboard-only navigation (<code>Tab</code> will jump to the next focusable element, or <code>Shift</code> + <code>Tab</code> to the previous).</p>
+
+<h3>Focus Order</h3>
+<p>It's important to keep focus behavior logical and predictable. This can require extra care when user interaction modifies the tabbable order of elements, such as by opening or closing a modal. In the case of a modal, focus should move to an element inside the modal while open (and not be permitted to reach the page below). Conversely, closing the modal should revert focus back to the originating element that triggered opening it. </p>
```
[the remark this answers is not in the document]
Comment (glunn) [reply app/templates/index.hbs]: Thanks @MelSumner, really appreciate the feedback!! I pushed up these updates. Is there anything specific you had in mind for the focus management in Ember section? Maybe something about route navigation?

[the remark this answers is not in the document]
Comment (MelSumner) [reply app/templates/index.hbs]: I think we can add that in a subsequent PR! 👍

[the remark this answers is not in the document]
Comment (glunn) [reply app/templates/index.hbs]: Sounds great!

**Pull request merged**